src/blog/2024/01/sentiment-analysis-with-node-red.md

```diff
@@ -0,0 +1,103 @@
+- **Campaign Monitoring:** Another use case of sentiment analysis is a measure of influence which is crucial in any marketing campaign.
+- **Brand Monitoring:** Brand monitoring is another great use case for sentiment analysis. Companies can use sentiment analysis to check the social media sentiments around their brand from their audience.
+
+# Building a Form in Dashboard 2.0
+
+In this system, we will analyse the sentiment of text content obtained from the user.  For this we are going to build a user interface using Dashboard 2.0 and Node-RED.
+
+1. Install Node-RED Dashboard 2.0. Follow these [instructions](https://dashboard.flowfuse.com/getting-started.html) to install.
+2. Create your first group, page, theme, and base.
+3. Drag a ui form widget to the canvas and select the created group.
+4. Add an element in the form widget and give it a name and label, select the type as multiline, and set the number of rows according to your need.
+
+!["Taking user input for Sentiment analysis using form"](./images/sentiment-analysis-form.png "Taking user input for Sentiment analysis using form")
+
+# Normalizing the data
+
+We need to normalize the payload before sending it to the next node because the form widget always returns an object containing the property of values of form elements.
+
+1. Drag a change node to canvas.
+2. Set `msg.payload.$FORM_ELEMENT_NAME` to `msg.payload`, replace the `$FORM_ELEMENT_NAME` with the name of the form element that you have added to the form to obtain user input.
+3. Connect the UI form nodes output to the change node’s input.
+
+!["Normalizing the payload using change node"](./images/sentiment-anlaysis-change-node(1).png "Normalizing the payload using change node")
+
+# Installing custom node
+
+Now it’s time to install a custom node that can perform sentiment analysis for us. In this guide, we will use the `node-red-node-sentiment` which is a Node-RED node that uses the AFINN-165 wordlists for sentiment analysis of words. It returns a sentiment object containing a score and other properties but we will only use the score property. Score property typically ranges from -5 to 5.
+
+1. Install the `node-red-node-sentiment` package by the node-red palette manager.
+2. Drag a sentiment node to canvas.
+3. Connect the change nodes output to sentiment node input.
+
+# Calculating percentage
+
+Why do we need to calculate the percentage? We will show the final result with the help of a circular progress bar and three different emojis. Ideally we should show the progress bar based on a percentage of score instead of negative values.
+
+1. Drag another change node to canvas.
+2. set `msg.payload` to `((msg.sentiment.score - (-5)) / (5 - (-5))) * 100` as a JSON expression, it will calculate the percentage of the score.
```

Comment:
```suggestion
2. set `msg.payload` to `((msg.sentiment.score - (-5)) / (5 - (-5))) * 100` as a JSONata expression, it will calculate the percentage of the score.
```